maat-court-data-api/src/test/java/gov/uk/courtdata/dces/controller/ContributionFileControllerTest.java

```diff
@@ -0,0 +1,135 @@
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@WebMvcTest(ContributionFileController.class)
+@AutoConfigureMockMvc(addFilters = false)
+class ContributionFileControllerTest {
+    private static final String BASE_URL = "/api/internal/v1/debt-collection-enforcement/contribution-file";
+
+    private final ContributionFileMapper mapper = new ContributionFileMapperImpl();
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private ContributionFileService contributionFileService;
+
+    @Test
+    void givenCorrectParameters_whenGetContributionFileInvoked_thenResponseIsReturned() throws Exception {
+        final int fileId = 99;
+        final var response = mapper.toContributionFileResponse(TestEntityDataBuilder.getPopulatedContributionFilesEntity(fileId));
+        when(contributionFileService.getContributionFile(fileId)).thenReturn(Optional.of(response));
+
+        mockMvc.perform(MockMvcRequestBuilders.get(BASE_URL + "/{fileId}", fileId))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.id").value(fileId));
+        verify(contributionFileService).getContributionFile(fileId);
+    }
+
+    @Test
+    void givenIncorrectParameters_whenGetContributionFileInvoked_thenNotFound() throws Exception {
+        final int INCORRECT_FILE_ID = 666;
+        when(contributionFileService.getContributionFile(INCORRECT_FILE_ID)).thenReturn(Optional.empty());
+
+        mockMvc.perform(MockMvcRequestBuilders.get(BASE_URL + "/{fileId}", INCORRECT_FILE_ID))
+                .andExpect(status().isNotFound());
+        verify(contributionFileService).getContributionFile(INCORRECT_FILE_ID);
+    }
+
+    @Test
+    void givenInvalidParameters_whenGetContributionFileInvoked_thenBadRequest() throws Exception {
+        final String INVALID_FILE_ID = "crouton";
+
+        mockMvc.perform(MockMvcRequestBuilders.get(BASE_URL + "/{fileId}", INVALID_FILE_ID))
+                .andExpect(status().isBadRequest());
+        verify(contributionFileService, never()).getContributionFile(anyInt());
+    }
+
+    @Test
+    void givenCorrectParameters_whenGetAllContributionFileErrorInvoked_thenResponseIsReturned() throws Exception {
+        final int fileId = 99, contributionId = 888;
```

Comment:
I might be wrong, but I think one of the Java style guidelines suggest declaring only a maximum of 1 variable per line.

Reply:
Found https://gds-way.digital.cabinet-office.gov.uk/manuals/programming-languages/java.html#java-style-guide which points to https://google.github.io/styleguide/javaguide.html#s4.8.2-variable-declarations and https://www.oracle.com/java/technologies/javase/codeconventions-declarations.html#2992 - so you're absolutely right.

Done.